eden/scm/edenscm/git.py

```diff
@@ -142,7 +143,24 @@ def clone(ui, url, destpath=None, update=True, pullnames=None):
 
                 if not nodes:
                     pullnames.append(update)
-
+            else:
+                b = callgitnorepo(ui, ["ls-remote", "--symref", url, "HEAD"])
+                def parse_symref_head(b):
+                    out = b.stdout.decode('utf-8')
+                    """
+                    ref: refs/heads/trunk   HEAD
+                    82156f7fd2719f5cca9f8a8738ff4e61561d04aa        HEAD
+                    """
+                    symrefhead = out.split("\n")[0]
+                    '''
+                    ref: refs/heads/trunk   HEAD
+                    '''
+                    headref = symrefhead.split(" ")[1].split("\t")[0]
+                    '''
+                    refs/heads/trunk
+                    '''
```

Comment:
nit: your comment style are inconsistent.  You sometimes use `"""` and sometimes use `'''`

Reply:
Good call out, will fix this

Reply:
fixed